js/ui/selection/selection.strategy.deferred.js

```diff
@@ -179,8 +179,22 @@ export default SelectionStrategy.inherit({
         selectionFilter = that._denormalizeFilter(selectionFilter);
 
         if(selectionFilter && selectionFilter.length) {
-            that._removeSameFilter(selectionFilter, filter, isDeselect, isSelectAll);
+            const removedIndex = that._removeSameFilter(selectionFilter, filter, isDeselect, isSelectAll);
             const filterIndex = that._removeSameFilter(selectionFilter, filter, !isDeselect);
+
+            const onlyNegativeFiltersLeft = selectionFilter.every((v, i) => {
```

Comment:
Extract onlyNegativeFiltersLeft calculation to separate method.
Also rename v to more understandable name(for example, filterItem).